lib/services/dialog_import_service.rb

```diff
@@ -99,12 +99,14 @@ def check_field_associations(fields)
   end
 
   def import(dialog)
-    @dialog_import_validator.determine_dialog_validity(dialog)
-    new_dialog = Dialog.create(dialog.except('dialog_tabs', 'export_version'))
-    association_list = build_association_list(dialog)
-    new_dialog.update!(dialog.merge('dialog_tabs' => build_dialog_tabs(dialog, dialog['export_version'] || DEFAULT_DIALOG_VERSION)))
-    build_associations(new_dialog, association_list)
-    new_dialog
+    ActiveRecord::Base.transaction do
+      @dialog_import_validator.determine_dialog_validity(dialog)
+      new_dialog = Dialog.create(dialog.except('dialog_tabs', 'export_version'))
```

Comment:
Is it only validation that is preventing the dialog from being create while the associated tables are being created, orphaned?

Reply:
in other words, where is it failing, on the `update!` below?

Reply:
I'm trying to understand where the associated tables are created but not the dialog itself.

There are several things we can do:
* use new
* call save! later
* only `update!` `if valid?`
...

Reply:
no, on the very line you're commenting on, the `create` due to the unique on name that's on https://github.com/ManageIQ/manageiq/blob/master/app/models/dialog.rb#L15

Reply:
Ultimately, we should call a method that will raise on exception before and during `build_associations` so the transaction gets rolled back.

Reply:
> no, on the very line you're commenting on, the `create` due to the unique on name that's on https://github.com/ManageIQ/manageiq/blob/master/app/models/dialog.rb#L15

Hmm, but `create` will not raise on validation failure, `create!` will.  Right?  Sometime later it might be raising, such as the `update!`

Reply:
yeah, fine, you're right, `new_dialog.update!(dialog.merge('dialog_tabs' => build_dialog_tabs(dialog, dialog['export_version'] || DEFAULT_DIALOG_VERSION)))`

Reply:
note: the `create` on the line from the original comment in this thread _cannot_ be a `create!` because then the UI will never let you save a dialog since it will have no tabs.